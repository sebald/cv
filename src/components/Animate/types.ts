import * as React from 'react';

export type AnimateProps = {
  className?: string;
};
