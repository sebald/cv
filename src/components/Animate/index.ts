export * from './Animate';
export * from './types';
