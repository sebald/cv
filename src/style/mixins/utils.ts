import { CSSProperties } from 'react';
import { StyledMixin } from '../theme';

export const propOrEmpty = <P extends keyof CSSProperties>(
  name: P
): StyledMixin<Pick<CSSProperties, P>> => (props: Pick<CSSProperties, P>) => {
  const p = props[name];
  return p ? { [name]: p } : {};
};
