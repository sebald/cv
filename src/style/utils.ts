import { theme } from './theme';

const { htmlFontSize } = theme;

export const px = (value: number) => `${value}px`;

export const pxToRem = (pixel: number, base = htmlFontSize) =>
  `${parseFloat((pixel / base).toFixed(5))}rem`;

export const remToPx = (rem: string, base = htmlFontSize) =>
  parseFloat(rem) * base;

export const ensurePx = (value: number | string, base = htmlFontSize) =>
  typeof value === 'string' ? remToPx(value, base) : value;
