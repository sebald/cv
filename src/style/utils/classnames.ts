export const mergeClassNames = (...names: (string | undefined)[]) => ({
  className: names
    .filter(x => typeof x === 'string')
    .join(' ')
    .trim(),
});
