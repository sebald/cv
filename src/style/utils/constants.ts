export const htmlFontSize = 16;

export const ratio = {
  perfectFourth: 1.333,
  perfectFifth: 1.5,
  goldenRatio: 1.618,
  majorSixth: 1.667,
};
