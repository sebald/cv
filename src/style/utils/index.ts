export * from './classnames';
export * from './constants';
export * from './createMixinComponent';
export * from './interpolation';
export * from './unit';
export * from './typography';
