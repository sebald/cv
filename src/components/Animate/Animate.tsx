import React from 'react';

import { animation, mergeClassNames, styled } from '../../style';
import { AnimateProps } from './types';

const AnimatedChild: React.SFC<AnimateProps> = ({ className, children }) => {
  const styledChildren = React.Children.toArray(children).map(
    child =>
      React.isValidElement<any>(child)
        ? React.cloneElement(
            child,
            mergeClassNames(child.props.className, className)
          )
        : child
  );

  return <>{styledChildren}</>;
};

export const Animate = styled(AnimatedChild)`
  ${animation};
`;
