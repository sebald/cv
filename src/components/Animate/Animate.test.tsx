import * as React from 'react';

import { mountWithTheme } from '../../test.utils';
import { Animate } from './Animate';

test('render correctly', () => {
  const wrapper = mountWithTheme(<Animate animation="fadeInUp" />);
  expect(wrapper).toMatchSnapshot();
});

test('add animation to children', () => {
  const wrapper = mountWithTheme(
    <Animate animation="fadeInUp" duration={1000}>
      <div>Hello</div>
      <h2>there</h2>
    </Animate>
  );
  expect(wrapper).toMatchSnapshot();
});
