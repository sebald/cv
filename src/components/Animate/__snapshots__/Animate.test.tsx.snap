// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add animation to children 1`] = `
.c0 {
  -webkit-animation: eqHJzO 1000ms cubic-bezier(0.4,0.0,0.2,1) 0ms none;
  animation: eqHJzO 1000ms cubic-bezier(0.4,0.0,0.2,1) 0ms none;
}

<Styled(StyledChildren)
  animation="fadeInUp"
  duration={1000}
>
  <StyledChildren
    animation="fadeInUp"
    className="c0"
    duration={1000}
  >
    <div
      className="c0"
      key=".0"
    >
      Hello
    </div>
    <h2
      className="c0"
      key=".1"
    >
      there
    </h2>
  </StyledChildren>
</Styled(StyledChildren)>
`;

exports[`render correctly 1`] = `
.c0 {
  -webkit-animation: eqHJzO 250ms cubic-bezier(0.4,0.0,0.2,1) 0ms none;
  animation: eqHJzO 250ms cubic-bezier(0.4,0.0,0.2,1) 0ms none;
}

<Styled(StyledChildren)
  animation="fadeInUp"
>
  <StyledChildren
    animation="fadeInUp"
    className="c0"
  />
</Styled(StyledChildren)>
`;
